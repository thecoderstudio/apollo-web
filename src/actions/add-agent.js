const SELECT_OPERATING_SYSTEM = 'SELECT_OPERATING_SYSTEM';
const SELECT_ARCHITECTURE = 'SELECT_ARCHITECTURE';

function selectOperatingSystem(operatingSystem) {
  return {
    type: SELECT_OPERATING_SYSTEM,
    selectedOperatingSystem: operatingSystem
  };
}

function selectArchitecture(architecture) {
  return {
    type: SELECT_ARCHITECTURE,
    selectedArchitecture: architecture
  };
}

export {
  selectOperatingSystem,
  SELECT_OPERATING_SYSTEM,
  selectArchitecture,
  SELECT_ARCHITECTURE
};
