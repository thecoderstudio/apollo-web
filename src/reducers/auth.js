import Cookies from 'js-cookie';

function logout() {
  Cookies.remove('AIOHTTP_SESSION', {path: '/'});
}

export default function authReducer(state = { authenticated: false }, action) {
  if (action.type == 'LOGOUT') {
    logout();
  }
  return { authenticated: Boolean(Cookies.get('AIOHTTP_SESSION', { path: '/' })) };
}
