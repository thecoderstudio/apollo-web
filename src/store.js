import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import storage from 'redux-persist/lib/storage';
import mapTransformer from './transformers/map';
import rootReducer from './reducers/root';

const persistConfig = {
  key: 'root',
  whitelist: ['auth', 'agent'],
  transforms: [mapTransformer({ whitelist: 'agent' })],
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
