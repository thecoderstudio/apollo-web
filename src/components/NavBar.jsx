import React from 'react';
import styled from 'styled-components';
import { logout as logoutAction } from '../actions/auth';
import Text from './Text';

const NavigationBar = styled.div`
  height: 50px;
  padding: 10px;
  display: grid;
  grid-template-columns: 1fr [options] 300px;

  background-color: ${props => props.theme.lightBlack};
`;

const Logout = styled(Text)`
  float: right;
  max-width: 300px;

  &:hover {
    cursor: pointer;
  }
`;

class NavBar extends React.PureComponent {
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  logout() {
    let { dispatch } = this.props;
    dispatch(logoutAction());
  }

  render() {
    return (
      <NavigationBar>
        <Logout onClick={this.toggleDropDown}>Logout</Logout>
      </NavigationBar>
    );
  }
}

export default  (NavBar);