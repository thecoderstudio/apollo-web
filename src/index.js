import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
