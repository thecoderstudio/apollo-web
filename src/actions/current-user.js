export const CACHE_CURRENT_USER = 'CACHE';
export const REMOVE_CURRENT_USER = 'REMOVE';
import { parseSnakeCaseObj } from '../util/parser';

function removeCurrentUser() {
  return {
    type: REMOVE_CURRENT_USER
  };
}

function cacheCurrentUser(user) {
  return {
    type: CACHE_CURRENT_USER,
    user: parseSnakeCaseObj(user)
  };
}

export { removeCurrentUser, cacheCurrentUser };
