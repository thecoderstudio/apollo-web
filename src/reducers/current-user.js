import { CACHE_CURRENT_USER, REMOVE_CURRENT_USER } from '../actions/current-user';

export default function currentUserReducer(state={}, action) {
  switch(action.type) {
    case CACHE_CURRENT_USER:
      return action.user;
    case REMOVE_CURRENT_USER:
      return {};
    default:
      return state;
  }
}
