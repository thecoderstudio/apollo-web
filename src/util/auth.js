import { store } from '../store';
import { logout } from '../actions/auth';
import { removeCurrentUser } from '../actions/current-user';

export function logout() {
  store.dispatch(logout());
  store.dispatch(removeCurrentUser());
}
