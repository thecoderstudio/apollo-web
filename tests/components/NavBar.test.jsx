import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import NavBar, { NavBar as PlainNavBar } from '../../src/components/NavBar';
import { store as globalStore } from '../../src/store';
import { logout as logoutAction } from '../../src/actions/auth';
import { removeCurrentUser } from '../../src/actions/current-user';

const mockStore = configureStore([]);

function getComponent(store) {
  return renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>
  );
}

describe('navbar', () => {
  let store;
  let dispatchSpy;

  beforeEach(() => {
    store = mockStore({
      currentUser: {}
    });
    dispatchSpy = jest.spyOn(globalStore, 'dispatch');
  });

  it('renders correctly', () => {
    const tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles successful logout', () => {
    const tree = getComponent(store);
    const instance = tree.root;
    instance.findByProps({ id: 'logoutButton' }).props.onClick();
    expect(dispatchSpy).toHaveBeenCalledWith(logoutAction());
    expect(dispatchSpy).toHaveBeenCalledWith(removeCurrentUser());
  });

  it('toggles collapsed', () => {
    const component = getComponent(store);
    const { root } = component;
    const toggle = root.findByProps({ id: 'collapseIcon' });
    const menu = root.findByType(PlainNavBar);
    toggle.props.onClick();
    component.toJSON();
    expect(menu.instance.state.collapsed).toBe(false);

    toggle.props.onClick();
    component.toJSON();
    expect(menu.instance.state.collapsed).toBe(true);
  });

  it('render admin link for admins', () => {
    store = mockStore({
      currentUser: {
        username: 'admin',
        role: {
          name: 'admin'
        }
      }
    });
    const tree = getComponent(store).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
