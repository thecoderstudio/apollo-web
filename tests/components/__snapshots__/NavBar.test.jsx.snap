// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar handles successful logout 1`] = `
.c3 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c3:hover {
  opacity: 0.90;
}

.c3:focus {
  outline: none;
}

.c3:active {
  opacity: 0.80;
}

.c3:disabled:hover {
  opacity: 1;
  cursor: auto;
}

.c4 {
  background-color: transparent;
  border: 1px solid;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  opacity: 0.90;
}

.c0 {
  height: 50px;
  padding: 16px;
  display: grid;
  grid-template-columns: 1fr 8fr [logout];
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-right: 16px;
}

.c5 {
  grid-column: logout;
  float: right;
  width: 100px;
  height: 75%;
}

<div
  className="c0"
>
  <h3>
    Apollo
  </h3>
  <div>
    <a
      className="c1 c2"
      href="/"
      onClick={[Function]}
    >
      Dashboard
    </a>
  </div>
  <button
    className="c3 c4 c5"
    onClick={[Function]}
  >
    Log out
  </button>
</div>
`;

exports[`nav bar render admin link for admins 1`] = `
.c3 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c3:hover {
  opacity: 0.90;
}

.c3:focus {
  outline: none;
}

.c3:active {
  opacity: 0.80;
}

.c3:disabled:hover {
  opacity: 1;
  cursor: auto;
}

.c4 {
  background-color: transparent;
  border: 1px solid;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  opacity: 0.90;
}

.c0 {
  height: 50px;
  padding: 16px;
  display: grid;
  grid-template-columns: 1fr 8fr [logout];
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-right: 16px;
}

.c5 {
  grid-column: logout;
  float: right;
  width: 100px;
  height: 75%;
}

<div
  className="c0"
>
  <h3>
    Apollo
  </h3>
  <div>
    <a
      className="c1 c2"
      href="/"
      onClick={[Function]}
    >
      Dashboard
    </a>
    <a
      className="c1 c2"
      href="/admin"
      onClick={[Function]}
    >
      Admin
    </a>
  </div>
  <button
    className="c3 c4 c5"
    onClick={[Function]}
  >
    Log out
  </button>
</div>
`;

exports[`nav bar renders correctly 1`] = `
.c3 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
}

.c3:hover {
  opacity: 0.90;
}

.c3:focus {
  outline: none;
}

.c3:active {
  opacity: 0.80;
}

.c3:disabled:hover {
  opacity: 1;
  cursor: auto;
}

.c4 {
  background-color: transparent;
  border: 1px solid;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  opacity: 0.90;
}

.c0 {
  height: 50px;
  padding: 16px;
  display: grid;
  grid-template-columns: 1fr 8fr [logout];
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-right: 16px;
}

.c5 {
  grid-column: logout;
  float: right;
  width: 100px;
  height: 75%;
}

<div
  className="c0"
>
  <h3>
    Apollo
  </h3>
  <div>
    <a
      className="c1 c2"
      href="/"
      onClick={[Function]}
    >
      Dashboard
    </a>
  </div>
  <button
    className="c3 c4 c5"
    onClick={[Function]}
  >
    Log out
  </button>
</div>
`;
